Introduce BUZZER on pin PB12 (GPIOB-12)

surfdado opened the pull request:
Trigger buzzer when above 1500 erpm and switch_state is OFF
Stop buzzer when switch is back ON, or when timer expires and
a fault is triggered.

TODO: make buzzer speed threshold configurable

Signed-off-by: Dado Mista <dadomista@gmail.com>

b264 commented on applications/app_balance.c:
```diff
@@ -412,7 +418,17 @@ static THD_FUNCTION(balance_thread, arg) {
+		else {
+			// if the switch comes back on we stop buzzing
+			BUZZER_OFF();
+		}
```
Do you think it's a better idea to encase this entire logic block in a `#ifndef` so that it's absent if compiled without the feature?

b264 commented on applications/app_balance.c:
```diff
@@ -64,6 +64,11 @@ typedef enum {
 	ON
 } SwitchState;
 
+#ifndef BUZZER_ON
```
Should this be named differently than the function name, maybe `FOOTSWITCH_BUZZER_ENABLE` or something?

surfdado commented:
I personally am not a big fan of ugly ifdefs in the middle of the code. I’ve structured it so that this compiles even if you don’t have the feature, e.g. for a different VESC. But to make it conditional it should be just a runtime configuration option, don’t you think? We don’t want firmware version with/without such a feature…

 

From: B <notifications@github.com> 
Sent: Sunday, November 15, 2020 9:15 AM
To: variESC/bldc <bldc@noreply.github.com>
Cc: surfdado <dadomista@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [variESC/bldc] Introduce BUZZER on pin PB12 (GPIOB-12) (#4)

 

@b264 requested changes on this pull request.

  _____  

In applications/app_balance.c <https://github.com/variESC/bldc/pull/4#discussion_r523784381> :

> +            if (switch_state == OFF) {
+                      if (abs_erpm > 1500)    // If we're above 5km/h and the switch is off => ALERT the user
+                              BUZZER_ON();
+                      else
+                              // if we drop below 5km/h stop buzzing
+                              BUZZER_OFF();
+              }
+              else {
+                      // if the switch comes back on we stop buzzing
+                      BUZZER_OFF();
+              }

Do you think it's a better idea to encase this entire logic block in a #ifndef so that it's absent if compiled without the feature?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/variESC/bldc/pull/4#pullrequestreview-530815986> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ARYNDWFK5PAKBISKBYJ6I4DSQAEBTANCNFSM4TWJHSFQ> .  <https://github.com/notifications/beacon/ARYNDWAQATJGFHKB3PTSST3SQAEBTA5CNFSM4TWJHSF2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOD6RZX4Q.gif>

surfdado commented:
Agreed

 

From: B <notifications@github.com> 
Sent: Sunday, November 15, 2020 9:16 AM
To: variESC/bldc <bldc@noreply.github.com>
Cc: surfdado <dadomista@gmail.com>; Author <author@noreply.github.com>
Subject: Re: [variESC/bldc] Introduce BUZZER on pin PB12 (GPIOB-12) (#4)

 

@b264 commented on this pull request.

  _____  

In applications/app_balance.c <https://github.com/variESC/bldc/pull/4#discussion_r523786517> :

> @@ -64,6 +64,11 @@ typedef enum {
        ON
 } SwitchState;
 
+#ifndef BUZZER_ON

Should this be named differently than the function name, maybe FOOTSWITCH_BUZZER_ENABLE or something?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/variESC/bldc/pull/4#pullrequestreview-530817380> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ARYNDWCQERQO5CTD2YZUDXDSQAEEJANCNFSM4TWJHSFQ> .  <https://github.com/notifications/beacon/ARYNDWDSRFENVCNZOAWFOS3SQAEEJA5CNFSM4TWJHSF2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOD6R2CZA.gif>

surfdado committed:
Balance: Introduce footswitch fault buzzer on pin PB12

Using Pin PB12 (aka GPIOB-12)

Trigger buzzer when above 1500 erpm and switch_state is OFF
Stop buzzer when switch is back ON, or when timer expires and
a fault is triggered.

Signed-off-by: Dado Mista <dadomista@gmail.com>

surfdado committed:
Balance: introduce directional light control

Pins PC13 and PC14 turn on/ when going fwd/backward respectively.
At speeds below the half-switch fault erpm both are kept on.

CAREFUL: do not drive powerful LEDs directly. Use a relais or MOSFET!

Signed-off-by: Dado Mista <dadomista@gmail.com>
